doc/LogitechSpotlightHID++.md

```diff
@@ -0,0 +1,137 @@
+
+The Projecteur further configure the Logitech device to send `Next Hold` and `Back Hold` events and reset the pointer speed to normal with following HID++ commands.
+
+```
+// enable next button hold (0x07 - Feature Index for ReprogramControlsV4 Feature Code, 0xda - next button, 0x33 - hold event)
+{0x11, 0x01, 0x07, 0x3d, 0x00, 0xda, 0x33, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00}
+// back button hold (0x07 - Feature Index for ReprogramControlsV4 Feature Code, 0xdc - back button, 0x33 - hold event)
+{0x11, 0x01, 0x07, 0x3d, 0x00, 0xdc, 0x33, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00}
+// Reset pointer Speed (0x0a - Feature Index for Reset Feature Code, 0x14 - 5th level pointer speeed (can be between 0x10-0x19))
+{0x10, 0x01, 0x0a, 0x1d, 0x14, 0x00, 0x00}
+```
+
+These initialization steps are implemented in `initialize` method of `SubHidppConnection` class in [device.h](../src/device.h). it should be noted that after the reprogramming of Next and Back button in first and second message, the spotlight device send relative mouse move data through HID++ messages (not regular input events to OS). These events are handled in `onHidppDataAvailable` method in `Spotlight` class in [spotlight.h](../src/spotlight.h).
+
+For the sake of completeness, it should be noted that the official Logitech Spotlight software reprogram the click and double click events too by following HID++ commands:
+
+```
+// Send click event as HID++ message (0x07 - Feature Index for ReprogramControlsV4 Feature Code, 0xd8-click button, 0x33- hold event)
+{0x11, 0x01, 0x07, 0x3d, 0x00, 0xd8, 0x33, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00}
+// Send double click as HID++ message (0x07 - Feature Index for ReprogramControlsV4 Feature Code, 0xdf - double click)
+{0x11, 0x01, 0x07, 0x3d, 0x00, 0xdf, 0x03, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00}
+```
+
+However, Projecteur does not send these packages. Instead it grab the events from mouse event device associated with Spotlight. The approach taken by Projecteur is advantageous as it help in implementing Input Mapping feature that official Logitech sofware lacks. However, this approach also make porting Projecteur to different platform somewhat difficult.
+
+# Wireless Notification on Activation/Deactivation of Spotlight Device
+
+The Spotlight device send wireless notification if it get activated. Wireless notification will be short HID++ message if Spotlight device is connected through USB dongle. Otherwise, it will be a long HID++ message.
+
+For short HID++ wireless notification, the third byte will be 0x41. In such message, 6th bit in 5th byte show activation status of spotlight device. If the 6th bit is 0 then device got active otherwise the device got deactivated.
```

Comment:
```suggestion
For short HID++ wireless notifications, the third byte will be `0x41`. In this message, the 6th bit in 5th byte shows the activation status of spotlight device. If the 6th bit is 0 then device got active, otherwise the device got deactivated.
```